zcu/encryption.py

```diff
@@ -20,27 +28,37 @@ def aes_decrypt(cipher, aes_key):
     encrypted_data = BytesIO()
     while True:
         aes_chunk = struct.unpack('>3I', cipher.read(12))
-        encrypted_data.write(cipher.read(aes_chunk[0]))
+        # aes_chunk[0] -> length not padded
+        # aes_chunk[1] -> padded length
+        if digi:
+          chunk_size=aes_chunk[1]
+        else:
+          chunk_size=aes_chunk[0]
+        encrypted_data.write(cipher.read(chunk_size))
         if aes_chunk[2] == 0:
             break
     encrypted_data.seek(0)
-    aes_cipher = AES.new(aes_key, AES.MODE_ECB)
+    if digi:
+        (k, v) = digi_key(aes_key)
+        aes_cipher = AES.new(k, AES.MODE_CBC, v[:16])
+    else:
+        aes_cipher = AES.new(aes_key, AES.MODE_ECB)
     decrypted_data = BytesIO()
     decrypted_data.write(aes_cipher.decrypt(encrypted_data.read()))
 
     decrypted_data.seek(0)
     return decrypted_data
 
 
-def aes_encrypt(infile, aes_key, chunk_size, include_unencrypted_length=False):
+def aes_encrypt(infile, aes_key, chunk_size, digi, include_unencrypted_length=False):
```

Comment:
digi=False

Reply:
changes, as well as in the aes_decrypt function